### PR TITLE
feat: add parity between ga4 and gtm for `main`

### PR DESCRIPTION
**Pull request by talbertosilva**
## Description

- Added changes to guarantee a better parity between ga4 and gtm tracking, this involves the addition of events that weren't being tracked on gtm and the respective parameters based on ga4 trackings.
- Changed needed utils functions that are directly related with the trackings as well.
- Implemented and adjusted the unit tests affected by this changes.

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/react/src/analytics/integrations/GTM/eventsMapper.ts [hunk 1]
```diff
@@ -1,15 +1,99 @@
...
+  utils,
 } from '@farfetch/blackout-analytics';
 import { get } from 'lodash-es';
 import { getEventProperties, getProductData } from './utils.js';
 import type { EventMappers } from './types/index.js';
 
 const noPropertiesMappedFn = () => ({});
 
+/**
+ * Returns the total event value for GA4 ecommerce events.
```
Comment (boliveira) [inline packages/react/src/analytics/integrations/GTM/eventsMapper.ts:17]: Fix this description please. Note that GTM does have the concept of "ecommerce" events like GA4 so I would change this description to be a bit more specific.

File: packages/react/src/analytics/integrations/GTM/eventsMapper.ts [hunk 2]
```diff
@@ -23,40 +107,64 @@ const eventsMapper: EventMappers = {
       products: get(properties, 'products', []).map(getProductData),
       list: properties.list,
       currency: properties.currency,
+      from: properties.from,
+      error: properties.error,
+      itemListId: properties.listId,
+      itemListName: properties.list,
```
Comment (boliveira) [inline packages/react/src/analytics/integrations/GTM/eventsMapper.ts:113]: We have `list` and `itemListName` being mapped to the same property. Can't we use just one of them?

File: packages/react/src/analytics/integrations/GTM/eventsMapper.ts [hunk 3]
```diff
@@ -23,40 +107,64 @@ const eventsMapper: EventMappers = {
...
     };
   },
   [EventType.ProductClicked]: (data: EventData<TrackTypesValues>) => {
     const properties = getEventProperties(data);
 
     return {
       product: getProductData(properties),
+      from: properties.from,
+      items: getProductItemsFromEvent(properties),
+      item_list_id: properties.listId,
```
Comment (boliveira) [inline packages/react/src/analytics/integrations/GTM/eventsMapper.ts:142]: Why some properties are written as `item_list_id` and others as `itemListId`? Same for `item_list_name` and `itemListName`.

File: packages/react/src/analytics/integrations/GTM/eventsMapper.ts [hunk 4]
```diff
@@ -142,8 +339,60 @@ const eventsMapper: EventMappers = {
...
+
+    return {
+      wishlistId: properties.wishlistId,
+    };
+  },
+  [PageType.Search]: (data: EventData<TrackTypesValues>) => {
+    const properties = getEventProperties(data);
+
+    return {
+      search_term: get(
```
Comment (boliveira) [inline packages/react/src/analytics/integrations/GTM/eventsMapper.ts:388]: Same as https://github.com/Farfetch/blackout/pull/977/files#r1470949119

Comment (talbertosilva) [reply packages/react/src/analytics/integrations/GTM/eventsMapper.ts]: Fixed! 👍🏼

File: packages/react/src/analytics/integrations/GTM/eventsMapper.ts (hunk 3 above)
Comment (talbertosilva) [reply packages/react/src/analytics/integrations/GTM/eventsMapper.ts]: Fixed! 👍🏼

File: packages/react/src/analytics/integrations/GTM/eventsMapper.ts (hunk 2 above)
Comment (talbertosilva) [reply packages/react/src/analytics/integrations/GTM/eventsMapper.ts:116]: Fixed! 👍🏼

File: packages/react/src/analytics/integrations/GTM/eventsMapper.ts (hunk 1 above)
Comment (talbertosilva) [reply packages/react/src/analytics/integrations/GTM/eventsMapper.ts]: Fixed! 👍🏼

**Review by boliveira — APPROVED** [review]

**Review by diegogriep — APPROVED** [review]

**Review by dnamorim — APPROVED** [review]

**Pull request merged**